### PR TITLE
create function to get ProcessGroupNCCL uid

### PR DESCRIPTION
**Pull request by shengbao-zheng**
Summary: expose ProcessGroupNCCL uid

Differential Revision: D54446056

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

### CODE REVIEW

File: test/distributed/test_c10d_nccl.py [hunk 1]
```diff
@@ -1524,6 +1524,18 @@ def test_non_blocking_with_eager_init(self):
         self.assertEqual(backend.comm_split_count(), 1)
 
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    def test_get_uid(self):
+        store = c10d.FileStore(self.file_name, self.world_size)
+        device = torch.device(f'cuda:{self.rank}')
+        pg = self._create_process_group_nccl(store, self.opts(), device_id=device)
```
Comment (aaronenyeshi) [inline test/distributed/test_c10d_nccl.py:1532]: does this test need to delete all existing pgs, if they were previously created by other unit tests?

**Review by aaronenyeshi — COMMENTED** [review]
Added a question

**Review by aaronenyeshi — CHANGES_REQUESTED** [review]
Also, please take a look at CI failures, and lint errors.

File: torch/distributed/distributed_c10d.py [hunk 2]
```diff
@@ -1072,6 +1073,16 @@ def get_backend(group: Optional[ProcessGroup] = None) -> Backend:
     pg_store = _world.pg_map[pg] if pg in _world.pg_map else None
     return Backend(not_none(pg_store)[0])
 
+def _get_process_group_uid(pg: ProcessGroup) -> int:
+    backend = None
+    try:
+        backend = pg._get_backend(torch.device("cuda"))
+    except RuntimeError:
+        pass
```
Comment (Skylion007) [inline torch/distributed/distributed_c10d.py:1081]: Shouldn't the exception at least be logged?

Comment (shengbao-zheng) [reply torch/distributed/distributed_c10d.py:1081]: All the RuntimeError exceptions in c10d during `_get_backend `are handled by 'pass' without logging. Just follow the same logic

File: test/distributed/test_c10d_nccl.py (hunk 1 above)
Comment (shengbao-zheng) [reply test/distributed/test_c10d_nccl.py:1532]: I think the testcase class has already helped clean pgs and each test will use a new tmp filestore.

**Review by aaronenyeshi — APPROVED** [review]
LGTM!

**Pull request closed without merging**